wordpress-mobile/WordPress-Android
# Issue/11799 show announcement on app upgrade

Fixes #11799

This PR adds logic for displaying feature announcements on the app upgrade.

We decided to show announcements independent of app version - eg. if you are on v4 of the app, and there is an announcement in v5, you will see even if you skip v5 and upgrade straight to v6 or v7 (until we remove or replace announcement manually).

To test (using wasabi build):
- If you install this branch on top of develop - the announcement should be displayed when app starts.
- If you perform the fresh installation, an announcement will not be displayed.
- To trigger an announcement bump the [versionCode](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/11799-show-announcement-on-app-upgrade/WordPress/src/main/java/org/wordpress/android/ui/whatsnew/FeatureAnnouncementProvider.kt#L8) number and restart the app.
- Make sure the announcement is not shown in vanilla version.



PR submission checklist:
